genie-proto/src/main/proto/genie.proto

```diff
@@ -145,6 +154,56 @@ message JobSpecificationServiceError {
 message JobSpecificationResponse {
     oneof response {
         JobSpecification specification = 1;
-        JobSpecificationServiceError error = 2;
+        JobSpecificationError error = 2;
+    }
+}
+
+message AgentMetadata {
+    string agent_version = 1;
+    string agent_hostname = 2;
+    string agent_pid = 3;
+}
+
+message ClaimJobRequest {
+    string id = 1;
+    AgentMetadata agent_metadata = 2;
+}
+
+message ClaimJobError {
+    enum Type {
+        UNKNOWN = 0;
+        ALREADY_CLAIMED = 1;
+        NO_SUCH_JOB = 2;
+    }
+    Type type = 1;
+    string message = 2;
+}
+
+message ClaimJobResponse {
+    oneof response {
+        bool successful = 1;
```

Comment:
Not a fan of having either a boolean (which supposedly is always true, if set) or an error message.
The boolean doesn't "add" any information, you could just check if the error is set or not.
Or have the boolean outside the `oneof` and have error set when it is false.

Reply:
Yes I agree. I couldn't think of a good way to do this. I had left it as `oneof` for consistency between APIs but I'm find with getting rid of it for some other mechanism. When you say "check if the error is set or not" do you mean the native gRPC error mechanism or just check if the error field is not null?

Reply:
The second, check if error is null.
Optionally keep the boolean `success` and the contract is that if that's set to false, then error is not going to be null.

Reply:
I like the second option there. I'll go with that